packages/components/src/TreeView/TreeView.component.js

```diff
@@ -44,51 +48,72 @@ import theme from './TreeView.scss';
  * <TreeView {...defaultProps} />
  *
  */
-const TreeView = ({
+function TreeView({
 	id,
 	headerText,
 	structure,
 	addAction,
 	addActionLabel,
 	itemSelectCallback,
 	itemToggleCallback,
-}) => (
-	<div className={theme['tc-treeview']}>
-		<header className={theme['tc-treeview-header']}>
-			<span>{headerText}</span>
-			{ addAction && <Action
-				label={addActionLabel}
-				icon="talend-plus"
-				onClick={addAction}
-				tooltipPlacement="right"
-				hideLabel
-				link
-				id={`${id}-add`}
-				key={addActionLabel}
-			/>}
-		</header>
-		<nav className={theme['tc-treeview-nav']}>
-			<ul className={theme['tc-treeview-ul']}>
-				{structure.map((item, i) => <TreeViewItem
-					id={id && `${id}-${i}`}
-					item={item}
-					itemSelectCallback={itemSelectCallback}
-					itemToggleCallback={itemToggleCallback}
-					key={i}
-				/>)}
-			</ul>
-		</nav>
-	</div>
-);
+	onClick,
+	onSelect,
+	noHeader,
+}) {
+	if (!cache.called && (itemSelectCallback || itemToggleCallback)) {
+		// eslint-disable-next-line
+		console.warn(`DEPRECATION WARNING: TreeView props migration:
+		itemToggleCallback -> onClick
+		itemSelectCallback -> onSelect`);
+		cache.called = true;
+	}
+	return (
+		<div className={theme['tc-treeview']}>
+			{!noHeader && (
+				<header className={theme['tc-treeview-header']}>
+					<span>{headerText}</span>
+					{addAction && (
+						<Action
+							label={addActionLabel}
+							icon="talend-plus"
+							onClick={addAction}
+							tooltipPlacement="right"
+							hideLabel
+							link
+							id={`${id}-add`}
+							key={addActionLabel}
+						/>
+					)}
+				</header>
+			)}
+			<nav className={theme['tc-treeview-nav']}>
+				<ul className={theme['tc-treeview-ul']}>
+					{structure.map((item, i) => (
+						<TreeViewItem
+							id={id && `${id}-${i}`}
+							item={item}
+							onSelect={onSelect || itemSelectCallback}
+							onClick={onClick || itemToggleCallback}
+							key={i}
+						/>
+					))}
+				</ul>
+			</nav>
+		</div>
+	);
+}
 
 TreeView.propTypes = {
 	id: PropTypes.string,
 	headerText: PropTypes.string,
-	structure: PropTypes.arrayOf(TreeViewItem.propTypes.item),
+	structure: PropTypes.arrayOf(PropTypes.object),
```

Comment:
Why change this type?

Reply:
because TreeViewItem import ../../index.js so it make circular deps and tests fails on a can't read propTypes of undefined.

Reply:
can you move proptypes in a place where it does not cause circular deps ?